deploy.py

```diff
@@ -493,6 +505,40 @@ def clear_apps(self):
       info(("A non-critical error occurred while erasing "
             "apps: {}".format(str(e))))
 
+  def clear_storage(self):
+    if self.args.programmer == "none":
+      return 0
+    board_props = SUPPORTED_BOARDS[self.args.board]
+    storage = bytes([0xFF] * board_props.storage_size)
+    # Use tockloader if possible
+    if self.args.programmer in ("jlink", "openocd"):
+      info("Erasing the persistent storage")
+      tock = loader.TockLoader(self.tockloader_default_args)
+      tock.open()
+      try:
+        tock.flash_binary(storage, board_props.storage_address)
+      except TockLoaderException as e:
+        fatal("Couldn't erase the persistent storage: {}".format(str(e)))
+      return 0
+    # Create an intelhex file otherwise
+    info("Creating the persistent storage HEX file")
+    # pylint: disable=g-import-not-at-top,import-outside-toplevel
+    import intelhex
+    storage_hex = intelhex.IntelHex()
+    storage_hex.frombytes(storage, offset=board_props.storage_address)
+    os.makedirs("target", exist_ok=True)
+    storage_file = "target/{}_storage.hex".format(self.args.board)
```

Comment:
This should rather be a tempfile:
```python
import tempfile
storage_file = tempfile.NamedTemporaryFile(delete=False)
storage_hex.tofile(storage_file, format="hex")
storage_file.close()
[...]
os.remove(storage_file.name)
```

Reply:
I wanted to do that first like in tockloader, but I've seen that they do a lot of weird stuff to support windows. However in your version you're fine keeping the file (`delete=False`) so I guess it would work around some issues. Anyway we don't need the file anymore.

Reply:
In normal mode, a tempfile is deleted when you close the file. Here, to avoid complex logic, we don't automatically delete the file but we delete it manually when we don't need it anymore (`os.remove()`)